File: csharp/lesson 2/ReverseList.cs
// https://leetcode.com/problems/reverse-linked-list

public class Solution 
{
    public ListNode ReverseList(ListNode head) 
    {
       ListNode prev = null, next = null, current = head;        
        while (current != null) 
        {
            next = current.next;
            current.next = prev;
            prev = current;
            current = next;
        }
        return prev; 
    }
}

File: python/lesson 2/reverse_list.py
from typing import Optional
from list_node import ListNode

# https://leetcode.com/problems/reverse-linked-list
class Solution:
    def reverse_list(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev = next = None
        current = head
        while current:
            next = current.next
            current.next = prev
            prev = current
            current = next
        return prev
